File: BankAccount.java
class BankAccount {
    private int numeroCuenta;
    private double saldoCuenta;
    private String tipoCuenta; 

    public BankAccount(int numeroCuenta, double saldoCuenta, String tipoCuenta) {
        this.numeroCuenta = numeroCuenta;
        this.saldoCuenta = saldoCuenta;
        this.tipoCuenta = tipoCuenta;
    }

    public int getNumeroCuenta() {
        return numeroCuenta;
    }

    public double getSaldoCuenta() {
        return saldoCuenta;
    }


    public void setSaldoCuenta(double saldoCuenta) {
        this.saldoCuenta = saldoCuenta;
    }

    public String getTipoCuenta(){
        return tipoCuenta;
    }

    public void depositA (String tipoCuenta, double saldoCuenta){
        if(tipoCuenta.equalsIgnoreCase("A")){

            if(saldoCuenta>50000){
                System.out.println("No se puede depositar más de 50,000 en una cuenta de tipo A");
                setSaldoCuenta(0);
            }else{
                setSaldoCuenta(saldoCuenta);
            }
        }
    }

    public void depositB (String tipoCuenta, double saldoCuenta){
        if(tipoCuenta.equalsIgnoreCase("B")){

            if(saldoCuenta>100000){
                System.out.println("No se puede depositar más de 100,000 en una cuenta de tipo B");
                setSaldoCuenta(0);
            }else{
                setSaldoCuenta(saldoCuenta);
            }
        }
    }

    
    public void depositC (String tipoCuenta, double saldoCuenta){
        if(tipoCuenta.equalsIgnoreCase("C")){
                setSaldoCuenta(saldoCuenta);
            
        }
    }
}

File: BankAccount.py
class BankAccount:
    def __init__(self, numero_cuenta, saldo_cuenta, tipo_cuenta):
        self.numero_cuenta = numero_cuenta
        self.saldo_cuenta = saldo_cuenta
        self.tipo_cuenta = tipo_cuenta

    def get_numero_cuenta(self):
        return self.numero_cuenta

    def get_saldo_cuenta(self):
        return self.saldo_cuenta

    def set_saldo_cuenta(self, saldo_cuenta):
        self.saldo_cuenta = saldo_cuenta

    def get_tipo_cuenta(self):
        return self.tipo_cuenta

    def depositA(self, tipo_cuenta, saldo_cuenta):
        if tipo_cuenta.upper() == "A":
            if saldo_cuenta > 50000:
                print("No se puede depositar más de 50,000 en una cuenta de tipo A")
                self.set_saldo_cuenta(0)
            else:
                self.set_saldo_cuenta(saldo_cuenta)

    def depositB(self, tipo_cuenta, saldo_cuenta):
        if tipo_cuenta.upper() == "B":
            if saldo_cuenta > 100000:
                print("No se puede depositar más de 100,000 en una cuenta de tipo B")
                self.set_saldo_cuenta(0)
            else:
                self.set_saldo_cuenta(saldo_cuenta)

    def depositC(self, tipo_cuenta, saldo_cuenta):
        if tipo_cuenta.upper() == "C":
            self.set_saldo_cuenta(saldo_cuenta)